### PR TITLE
[[ Merge 6.7.6 rc 1]]

### PR DESCRIPTION
**Pull request by livecodesebastien**
The only non-reviewed are the changes to the player, with had to be refactored.

### CODE REVIEW

**Comment by runrevmark** [comment]
There's a problem here due to subclassing a QTKit class - we can't do that as we mustn't link to QTKit directly.

File: engine/src/stack.cpp [hunk 1]
```diff
@@ -3133,6 +3124,28 @@ bool MCStack::resolve_relative_path(MCStringRef p_path, MCStringRef& r_resolved)
...
+
+    if (MCStringBeginsWith(p_path, MCSTR("./"), kMCCompareExact)
+            || MCStringBeginsWith(p_path, MCSTR(".\\"), kMCCompareExact))
+        t_start_index = 2;
+    else
+        t_start_index = 0;
+	
+    MCS_getcurdir(&t_cur_dir);
+
+    return MCStringFormat(r_resolved, "%@/%*@", *t_cur_dir, MCRangeMake(t_start_index, MCStringGetLength(p_path) - t_start_index), p_path);
```
Comment (livecodeali) [inline engine/src/stack.cpp]: The range argument here needs to be a pointer to an MCRange rather than an MCRange.

**Comment by livecodeali** [comment]
@livecode-vulcan review ok f11fec6

**Comment by livecodeali** [comment]
@livecode-vulcan review ok fa20608

**Comment by livecode-vulcan** [comment]
💙 review by @livecodeali ok fa20608a88ede892658a716cdbfefdcae22feb1b

**Pull request merged**